corpus/accounts/models.py

```diff
@@ -40,15 +41,111 @@ def users(self):
 
 
 class User(AbstractUser):
+    GENDERS = [
+        ("M", "Male"),
+        ("F", "Female"),
+        ("O", "Other"),
+        ("N", "Prefer not to disclose"),
+    ]
+
     username = None
-    phone_no = models.CharField(max_length=13, unique=True, blank=False, null=False)
-    gender = models.CharField(max_length=1, choices=GENDERS, blank=False, null=False)
-    email = models.EmailField(unique=True, blank=False, null=False)
+    phone_no = models.CharField(
+        max_length=15,
+        unique=True,
+        verbose_name="Phone Number",
+        validators=[validate_phone_number],
+    )
+    gender = models.CharField(max_length=1, choices=GENDERS)
+    email = models.EmailField(unique=True, verbose_name="Personal Email")
 
     USERNAME_FIELD = "email"
     REQUIRED_FIELDS = []
 
     objects = UserManager()
 
+    def save(self, *args, **kwargs):
+        phone = str(self.phone_no)
+        if len(phone) == 12 and phone.startswith("91"):
+            self.phone_no = phone[2:]
+        elif len(phone) == 13 and phone.startswith("+91"):
+            self.phone_no = phone[3:]
+
+        super(User, self).save(*args, **kwargs)
+
     def __str__(self):
         return f"{self.first_name} {self.last_name}"
+
+
+class ExecutiveMember(models.Model):
+    NITK_DEPARTMENTS = [
+        ("WO", "Water Resources and Ocean Engineering"),
+        ("CH", "Chemical Engineering"),
+        ("CM", "Chemistry"),
+        ("CV", "Civil Engineering"),
+        ("CS", "Computer Science and Engineering"),
+        ("EE", "Electrical and Electronics Engineering"),
+        ("EC", "Electronics and Communication Engineering"),
+        ("IT", "Information Technology"),
+        ("MA", "Mathematical and Computational Sciences"),
+        ("ME", "Mechanical Engineering"),
+        ("MT", "Metallurgical and Materials Engineering"),
+        ("MN", "Mining Engineering"),
+        ("PS", "Physics"),
+        ("SM", "School of Humanities, Social Sciences and Management"),
+    ]
+
+    user = models.OneToOneField(User, on_delete=models.CASCADE, primary_key=True)
+    sig = models.ForeignKey(SIG, on_delete=models.CASCADE)
+    edu_email = models.EmailField(
+        unique=True, verbose_name="NITK EDU Email", validators=[validate_nitk_email]
+    )
+    roll_number = models.CharField(
+        max_length=8,
+        unique=True,
+        verbose_name="NITK Roll Number",
+        validators=[validate_roll_number],
+    )
+    reg_number = models.CharField(
+        max_length=10,
+        unique=True,
+        verbose_name="NITK Registration Number",
+        validators=[validate_reg_number],
+    )
+    minor_branch = models.CharField(
+        max_length=2,
+        choices=NITK_DEPARTMENTS,
+        blank=True,
+        null=True,
+        verbose_name="Minor Branch",
+    )
+    ieee_number = models.CharField(
+        max_length=10,
+        blank=True,
+        null=True,
```

Comment:
Should make this `False`, as all exec members will have IEEE Membership number.

Reply:
They won't have membership numbers at the time of account creation. We first recruit people, then ask them to take up student membership.